Give a chance to add local rpms

qzheng527 opened the pull request:
Many users don't reply on the koji to generate local
packages. They usually put the local packages in some
local place. Once define LOCAL_RPM_PATH, copy the rpms
in this path as local rpms.

Signed-off-by: Qi Zheng <qi.zheng@intel.com>

mbelluzzo commented:
Hi @qzheng527 ,
We do realize that many users won't even want to use Koji for that purpose. For that reason, the code was structured to allow users to fit different "Content Providers" which allows to easily plug-and-play different ways to feed RPMs into the process.

So adding local rpms is a valid use case, but it would be better if structure in the form of a Content Provider.

Did you have a change to take a look at?
https://github.com/clearlinux/clr-distro-factory/wiki/Release#setup-content-content

The documentation isn't very detailed but gives the overall idea (patches for the docs are also welcome :-) ). Code wise the idea is to avoid 'content.sh' to become a big piece of code trying to fulfil every possible way to fetch RPM: ftp, ssh, local, koji, external repo, git LFS, http and so on.

The Pipeline file in the repo will never be a 1 size fits all. So it is a suggested Pipeline. The idea is that users would have their own pipeline and adding "content provider" steps before 'content.sh'. 

So it could be:
```
stage('Content') {
   steps {
      koji.sh
      local.sh
      ftp.sh
      content.sh
   }
}
```
Your proposal of local rpm is even a good content provider to be the default one on ReleasePipeline and be part of the 'make release' path, even that doing no-op for new users.

mbelluzzo commented on release/content.sh:
```diff
@@ -17,6 +17,15 @@ SCRIPT_DIR=$(dirname "$(realpath "${BASH_SOURCE[0]}")")
+if [[ -n ${LOCAL_RPM_PATH} && -d ${LOCAL_RPM_PATH} ]]; then
+    info "Copying local packages from ${LOCAL_RPM_PATH}"
+    cp "${LOCAL_RPM_PATH}"/*.rpm "${PKGS_DIR}"/
+    for file in "${LOCAL_RPM_PATH}"/*.rpm; do
```
Instead of a loop, why not to just `cp <path>/*.rpm <dest>/`?

If you just want to list the rpms on the console output for tracking (which is nice if is not 800 or so), you can echo the package list file that needs to be created to make content.sh. 

content.sh is expecting only two things of content provides:
 * a txt file inside ${WORK_DIR} following this name pattern ${PKG_LIST_FILE}_<content provider name> (no extension)
 * RPMs to be placed inside ${PKGS_DIR}

Conflicts, overrides etc are not handled atm. It is the user's job to ensure that a valid set of RPMs is inside ${PKGS_DIR} folder.

Anyhow, there is lots of room for improvements in this segment of code. This is definitely a step in this direction.

mbelluzzo reviewed (changes requested):
Plz see comments

Commit:
Add local rpm as content provider

Many users don't reply on the koji to generate local
packages. They usually put the local packages in some
local place. Once define LOCAL_RPM_PATH, copy the rpms
in this path as local rpms.

Signed-off-by: Qi Zheng <qi.zheng@intel.com>

Commit:
Add local into default release pipeline

Signed-off-by: Qi Zheng <qi.zheng@intel.com>

qzheng527 commented:
@mbelluzzo , please help review the new patches, thanks.

mbelluzzo commented:
Hey @qzheng527 there was just a few corner cases that I fixed and merged. Thank you!
